File: LeetCode/Medium/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.go
func minOperations(nums []int, k int) int {
    curr := nums[0]
    ret := 0

    for i := 1; i < len(nums); i++ {
        curr ^= nums[i]
    }

    for curr > 0 || k > 0 {
        if curr % 2 != k % 2 { ret++ }
        curr >>= 1
        k >>= 1
    }

    return ret
}

File: LeetCode/Medium/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k/2997-minimum-number-of-operations-to-make-array-xor-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int curr = nums[0];
        int ret = 0;
        
        for (int i = 1; i < nums.length; ++i)
            curr ^= nums[i];
        
        while (curr > 0 || k > 0){
            if (curr % 2 != k % 2) ++ret;
            curr >>= 1;
            k >>= 1;
        }
            
        return ret;
    }
}